test/test_empty_list_type.ml

```diff
@@ -0,0 +1,16 @@
+open Test_util
+
+let%expect_test "test_empty_list_type.beam" =
+  print_ast "test_empty_list_type.beam";
+  [%expect {|
+    (Ok (
+      AbstractCode (
+        ModDecl (
+          (AttrFile 1 test_empty_list_type.erl 1)
+          (AttrMod 1 test_empty_list_type)
+          (AttrExportType 3 (
+            (t 0)
+            (u 0)))
+          (DeclType 6 t () (TyPredef 6 nil ()))
```

Comment:
memo: 
In [the document](http://erlang.org/doc/apps/erts/absform.html), 
> the empty list type [] cannot be distinguished from the predefined type nil().

so `[]` type can be considered as predefined `nil()` type.